src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -0,0 +1,382 @@
+   *
+   * @param value        Data point value
+   * @param timestamp    Data point time stamp
+   * @param escapeQuotes Indicator to include escape characters before quotes
+   * @return
+   */
+  private static String buildBody(String value, String timestamp, boolean escapeQuotes) {
+    String quote;
+    if (escapeQuotes) {
+      quote = "\\\"";
+    } else {
+      quote = "\"";
+    }
+
+    String jsonBody =
+        "{"
+            + quote
+            + "Timestamp"
+            + quote
+            + ": "
+            + quote
+            + timestamp
+            + quote
+            + ","
+            + quote
+            + "Value"
+            + quote
+            + ": "
+            + value
+            + ","
+            + quote
+            + "UnitsAbbreviation"
+            + quote
+            + ": "
+            + quote
+            + quote
+            + ","
+            + quote
+            + "Good"
+            + quote
+            + ": true,"
+            + quote
+            + "Questionable"
+            + quote
+            + ": false"
+            + "}";
+    return jsonBody;
+  }
+
+  /**
+   * Builds the JSON contents for the end of the payload.
+   *
+   * @param timestamp The time should be in format yyyy-mm-ddThh:mm:ssZ.
+   * @return the payload segment as a string
+   */
+  public static String endPayload(String timestamp) {
+    String payload = "],\"lastSynchroDate\": \"" + timestamp + "\"}";
+    return payload;
+  }
+
+  public static String startOMFDataMessage() {
+    String payload = "[";
+    return payload;
+  }
+
+  /**
+   * * This constructs the start of a new container
+   *
+   * @param the payload segment as a string
+   */
+  public static String addContainerStartToOMFDataMessage(String tagName) {
+    // each tag's container id is set to the tag's name
+    String payload = "{"
+        + "\"containerid\": \"" + tagName + "\""
+        + ","
+        + "\"values\": [";
+
+    return payload;
+  }
+
+  /**
+   * This constructs a new data point omf message
+   *
+   * @return return the JSON payload for a new OMF data point
+   */
+  public static String addPointToOMFDataMessage(
+      String tagValue, String timestamp, boolean isFirstPoint) {
+    String payload = "";
+
+    if (!isFirstPoint) {
+      payload += ",";
+    }
+
+    payload += "{"
+        + " \"timestamp\": \""
+        + timestamp
+        + ".000Z\","
+        + " \"tagValue\": \"" + tagValue + "\""
+        + "}";
+
+    return payload;
+  }
+
+  /**
+   * call after finished adding all data points to close values array in json
+   *
+   * @return return the JSON payload for the end of an OMF container message.
+   */
+  public static String addContainerEndToOMFDataMessage() {
+    String payload =
+        "]}"; // closes off the end of the container specific portion of the data message
+    return payload;
+  }
+
+  /**
+   * call this when finished building up the OMF message
```

Comment:
```suggestion
   * Call this when finished building up the OMF message
```

Reply:
These "Call after/this" comments are pretty weak in terms of documentation. Its enough to just document what the method does.

Reply:
removing/changing these